src/test/java/com/wikia/webdriver/common/core/networktrafficinterceptor/NetworkTrafficInterceptor.java

```diff
@@ -54,7 +57,36 @@ public void checkAssetsStatuses(String domain) {
     har = getHar();
     for (HarEntry entry : har.getLog().getEntries()) {
       if (entry.getRequest().getUrl().contains(domain)) {
-        PageObjectLogging.log("RESPONSE STATUS: " + entry.getResponse().getStatus(), entry.getRequest().getUrl(), entry.getResponse().getStatus() < 400);
+        PageObjectLogging.log("RESPONSE STATUS: " + entry.getResponse().getStatus(), entry
+            .getRequest().getUrl(), entry.getResponse().getStatus() < 400);
+      }
+    }
+  }
+
+  /**
+   * Looks for correlator pattern in requests query strings to DFP domain, and logs if all the calls
+   * have the same correlator ID. Any difference is ID is logged as failure.
+   */
+  public void logDFP() {
+    har = getHar();
+    String expectedCorrelator = null;
+    Pattern pt = Pattern.compile("(correlator=)\\d*");
+
+    for (HarEntry entry : har.getLog().getEntries()) {
+      if (entry.getRequest().getUrl().contains("pubads.g.doubleclick.net")) {
+        Matcher matcher = pt.matcher(entry.getRequest().getQueryString().toString());
+        if (matcher.find()) {
+          String correlatorID = matcher.group(0);
+
+          if (expectedCorrelator == null) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)